messaging/core_test_utils.go

```diff
@@ -0,0 +1,129 @@
+package messaging
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/messaging/types"
+	wakutypes "github.com/status-im/status-go/waku/types"
+	"github.com/status-im/status-go/wakuv2"
+)
+
+type TestMessagingEnvironment struct {
+	// To enable communication between multiple messaging core instances in tests,
+	// share a single Waku instance across them.
+	waku *testWakuWrapper
+}
+
+func NewTestMessagingEnvironment() (*TestMessagingEnvironment, error) {
+	waku, err := newTestWakuWrapper()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
If this is only running in the `testing` context, then maybe pass `*testing.T` to it?

```suggestion
	waku, err := newTestWakuWrapper()
	require.NoError(t, err)
```

Reply:
Good point, but since it is used in one place atm, I'll keep simple `err` for now :pray: